control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -129,6 +129,15 @@ func (r *GatewayController) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
+	// fetch our inline certificates from cache, this needs to happen
+	// here since the certificates need to be reference counted before
+	// the gateways.
+	r.fetchConsulInlineCertificates(resources)
+
+	// add our current gateway even if it's not controlled by us so we
+	// can garbage collect any resources for it.
+	resources.ReferenceCountGateway(gateway)
```

Comment:
this allows us to garbage collect resources when something happens like a gateway switches gateway classes and is no longer controlled by us since the fetch below this only fetches things that we control